modules/search-provision/src/main/scala/io/renku/search/provision/handler/MessageReader.scala

```diff
@@ -59,18 +55,21 @@ object MessageReader:
   /** MessageReader that dequeues messages attempt to decode it. If decoding fails, the
     * message is marked as processed and the next message is read.
     */
-  def apply[F[_]: Sync](
+  def apply[F[_]: Async](
       queueClient: Resource[F, QueueClient[F]],
       queue: QueueName,
       clientId: ClientId,
-      chunkSize: Int
+      chunkSize: Int,
+      connectionRefresh: FiniteDuration
   ): MessageReader[F] =
     new MessageReader[F]:
       val logger = scribe.cats.effect[F]
 
       def read[A](using QueueMessageDecoder[F, A], Show[A]): Stream[F, Message[A]] =
-        for {
-          client <- Stream.resource(queueClient)
+        val s = for {
+          client <- Stream
+            .resource[F, QueueClient[F]](queueClient)
+            .interruptAfter(connectionRefresh)
```

Comment:
As discussed we could perhaps move this into the `QueueClient` ctor to create an infinite stream right away